dev/Deprecation.php

```diff
@@ -116,20 +126,47 @@ protected static function get_called_method_from_trace($backtrace, $level = 1) {
 			return $called['function'];
 		}
 	}
+	
+	/**
+	 * Determine if deprecation notices should be displayed
+	 * 
+	 * @return bool
+	 */
+	public static function get_enabled() {
+		// Deprecation is only available on dev
+		if(!Director::isDev()) {
+			return false;
+		}
+		if(isset(self::$enabled)) {
+			return self::$enabled;
+		}
+		if(defined('SS_DEPRECATION_ENABLED')) {
```

Comment:
Should the static var override the constant? I generally the constants take absolute precedence usually.

However, I'd suggest they have equal weight. if either is disabled, then don't show? Maybe that makes it a bit less flexible, though...